docs added for entry point gas estimations package (#233)
bcnmy/docs: docs/GasEstimations/index.mdx

---
slug: /gas-estimations
sidebar_position: 6
title: Gas Estimations
---

> While sending a userOp, It is essential to accurately estimate callGasLimit, verificationGasLimit and preVerficationGas. This ensures the userOp is included on the chain.
> The exact definition of these values is mentioned in the ERC 4337 spec [here](https://eips.ethereum.org/EIPS/eip-4337#useroperation).
> 
> Accurately estimating gas limits is crucial, as failure to do so may result in the following scenarios:
> 
> - If preVerfiicationGas is not enough the userOp can be rejected by bundlers
> - If callGasLimit is not enough the transaction will be included on chain but the call data execution will not be completed and the paymaster or account will still pay for the userOp
> - If verificationGasLimit is low the userOp will revert to the simulation step
> 
> All the above problems result in bad UX, failed and reverted transactions which are also hard to debug.
> The Gas estimation SDK enables you to estimate gas limits accurately to avoid such edge cases. This functionality is packaged within an npm package, enabling other infrastructure providers to reuse the same logic.

Reviewer: ```suggestion
While sending a userOp, It is essential to accurately estimate callGasLimit, verificationGasLimit and preVerficationGas. This ensures the userOp is included on the chain.
The exact definition of these values is mentioned in the ERC 4337 spec [here](https://eips.ethereum.org/EIPS/eip-4337#useroperation).

Accurately estimating gas limits is crucial, as failure to do so may result in the following scenarios:

- If preVerfiicationGas is not enough the userOp can be rejected by bundlers
- If callGasLimit is not enough the transaction will be included on-chain but the call data execution will not be completed and the paymaster or account will still pay for the userOp
- If verificationGasLimit is low the userOp will revert to the simulation step

All the above problems result in bad UX, failed and reverted transactions which are also hard to debug.
The Gas estimation SDK enables you to estimate gas limits accurately to avoid such edge cases. This functionality is packaged within an npm package, enabling other infrastructure providers to reuse the same logic.
```

---
slug: /gas-estimations
sidebar_position: 6
title: Gas Estimations
---

> While sending a userOp callGasLimit, verificationGasLimit and preVerficationGas have to be estimated accurately to get the userOp be included on the chain.

Reviewer 2: ```suggestion
While sending a userOp, It is essential to accurately estimate callGasLimit, verificationGasLimit and preVerficationGas. This ensures the userOp be included on the chain.
```

> While sending a userOp, It is essential to accurately estimate callGasLimit, verificationGasLimit and preVerficationGas. This ensures the userOp be included on the chain.
> The exact definition of these values are mentioned in the ERC 4337 spec [here](https://eips.ethereum.org/EIPS/eip-4337#useroperation).

Reviewer: ```suggestion
While sending a userOp, It is essential to accurately estimate callGasLimit, verificationGasLimit and preVerficationGas. This ensures the userOp is included on the chain.
The exact definition of these values is mentioned in the ERC 4337 spec [here](https://eips.ethereum.org/EIPS/eip-4337#useroperation).
```

> Accurately estimating them is very important because if not the following can happen:

Reviewer 2: ```suggestion
Accurately estimating gas limits is crucial, as failure to do so may result in the following scenarios:
```

> Accurately estimating gas limits is crucial, as failure to do so may result in the following scenarios:
> 
> - If preVerfiicationGas is not enough the userOp can be rejected by bundlers
> - If callGasLimit is not enough the transaction will be inculded on chain but the the call data execution will not be completed and the paymater or account will still pay for the userOp
> - If verificationGasLimit is low the userOp will revert in simulation step

Reviewer: ```suggestion
Accurately estimating gas limits is crucial, as failure to do so may result in the following scenarios:

- If preVerfiicationGas is not enough the userOp can be rejected by bundlers
- If callGasLimit is not enough the transaction will be included on chain but the call data execution will not be completed and the paymaster or account will still pay for the userOp
- If verificationGasLimit is low the userOp will revert to the simulation step
```

> All the above problems result in bad UX, failed and reverted transactions which also are sometimes ahrd to debug.

Reviewer 2: ```suggestion
All the above problems result in bad UX, failed and reverted transactions which are also hard to debug.
```

> We at biconomy have come up with ways estimate gas limits accurately so that such edge cases don't occur. All this is done in an npm package so that any other infra provider can resuse the same logic on their end.

Reviewer 2: ```suggestion
The Gas estimation SDK enables you to estimate gas limits accurately to avoid such edge cases. This functionality is packaged within an npm package, enabling other infrastructure providers to reuse the same logic.
```

> To dive deep into the workings of the contracts and logic of the package, please read this blog. ADD BLOG LINK

Reviewer 2: link to be added
